dashboard/app/controllers/api/v1/pd/application/principal_approval_applications_controller.rb

```diff
@@ -0,0 +1,11 @@
+module Api::V1::Pd::Application
+  class PrincipalApprovalApplicationsController < Api::V1::Pd::FormsController
+    def new_form
+      @application = Pd::Application::PrincipalApproval1819Application.new(
+        # current_user may not exist but might as well record it
+        user: current_user,
+        application_guid: params.try(:[], :application_guid)
```

Comment:
Application guid is required in the route: https://github.com/code-dot-org/code-dot-org/blob/560a83bb16690e7a7bd941aa252faafa132dab8d/dashboard/config/routes.rb#L432

We don't need to `try` here. You could even use `params.require(:application_guid)`